doc/SPIRE101.md

```diff
@@ -0,0 +1,96 @@
+
+
+# Overview
+
+Follow these walkthrough to set up your docker development environment and expose the Workload API. A workload client cmdline simulates a workload fetching SVIDs for its registered SPIFFE IDs.
+ 
+ ![cache manager flow](images/SPIRE101.png)
+
+### Requirement(s)
+
+#### GitHub Setup
+
+Clone the SPIRE github repo:
+
+    `git clone https://github.com/spiffe/spire`
+
+Note: If you don’t already have a GitHub account, please get setup[here](https://github.com).
+
+#### Docker Setup
+
+Pull the latest SPIRE image from spiffe/spire-dev docker repository:
+
+    `docker pull spiffe/spire-dev`
+Note: If you don't already have Docker already installed, please follow these[installation instructions](https://docs.docker.com/engine/installation/) .
+
+### Terminology
+
+ |Term                   | Description                                                      |
+ |-----------------------|------------------------------------------------------------------|
+ |spire-server           |  SPIRE server executable                                         |
+ |spire-agent            |  SPIRE agent executable                                          |
+ |wlcli                  |  Workload API GRPC client executable                             |
+ |socketPath             |  Unix Domain Socket file path through which workloads            |
+ |                       |  connects to Workload API                                        |                                          
+ |joinToken              |  Nonce generated by the SPIRE server to attest SPIRE agents      |
+
+
+### Walkthrough
+
+Setup your local Docker development environment, cd to the directory you cloned the SPIRE repo into and start the container 
+
+    cd ~/spire
+    docker-compose up -d
+
+Connect to the spire-dev container you just started 
+	
+    docker-compose exec spire-dev bash
+
+Create a user with uid 1000 for workload attestation
+	
+	useradd -u 1000 workload
+
+Install external dependencies by running
+	
+	make vendor
+
+The build target builds spire-server, spire-agent and wlcli executable, it also builds the plugin binaries.
+    
+    make build
+
+The `spire-server` and `spire-agent` executables along with their respective subcommands have a `—help`  option that give more details on using the specific cli options. Try running help for register sub command:
+	
+	./cmd/spire-server/spire-server register --help
+
+Familiarize with the server config
+	
+	cat conf/server/server.conf
+
```

Comment:
you should show the actual file here also.  Just copy an example in place.Have a link for the documentation also.